test/content/support.js

```diff
@@ -241,4 +241,26 @@ function resetDB() {
 	}).then(function() {
 		return Zotero.Schema.schemaUpdatePromise;
 	});
-}
\ No newline at end of file
+}
+
+
+/**
+ * Imports an attachment and calls a function when it is ready.
+ * @param {string} filename - The filename to import (in data directory)
+ * @param {function} f - A function that takes on argument, the id of the item.
+ * @returns {Promise}
+ */
+function* withImportedAttachmentFromFile(filename, f) {
```

Comment:
I think you need to wrap this in `Zotero.Promise.coroutine`, otherwise calling this function just returns the generator object, not the promise you expect. Don't see anything about Mocha/Chai being able to handle that (maybe I'm missing something). Try to make one of your tests fail, I'm fairly sure that it will still pass.